src/rgb/schema/transition.rs

```diff
@@ -28,7 +28,7 @@ use crate::csv::{serialize::Commitment, Error};
 #[display_from(Debug)]
 pub struct Transition {
     pub closes: Option<HashMap<usize, Occurences<u32>>>,
-    pub fields: Vec<Field>,
+    pub fields: HashMap<usize, Field>,
```

Comment:
I am very worried about custom-sized fields within consensus-critical data. Peter Todd review suggestion was to avoid them at all cost. So we need a fixed size here. BTW, can you pls explain rationale of changing Vec to HashMap?

Reply:
what do you mean custom-sized? this is pretty much equivalent to a vec of tuples, with the added bonus that you have indexing on the key.

the idea is that i need to know what the format of a field should be given a "type". before it was just a vec, so let's say that i'm validating the fields and i have a "type 0" that is a string, but multiple string fields defined in the schema. how do match them?

with this when you define the schema you say explicitly that the field of "type 0 should be a string", so when you have a transition that has a field of type 0 you know exactly how to validate it

Reply:
ah you mean to avoid `usize`? sure, we can do that

Reply:
yeah, `usize` thing, it's platform-specific so two apps compiled on different CPUs communicating over network or transferring files to each other will read messaged/data differently